verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSAllProfileResourceRecordSetApi.java

```diff
@@ -0,0 +1,127 @@
+package denominator.verisignmdns;
+
+import static denominator.common.Preconditions.checkArgument;
+import static denominator.common.Preconditions.checkNotNull;
+import static denominator.model.ResourceRecordSets.notNull;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import denominator.AllProfileResourceRecordSetApi;
+import denominator.Credentials;
+import denominator.common.Filter;
+import denominator.model.ResourceRecordSet;
+import denominator.model.Zone;
+import denominator.verisignmdns.VrsnMdns.Record;
+
+public class VerisignMDNSAllProfileResourceRecordSetApi implements denominator.AllProfileResourceRecordSetApi {
+    private final String domainName;
+    private final VrsnMdns api;
+    private final Provider<Credentials> credentialsProvider;
+
+    VerisignMDNSAllProfileResourceRecordSetApi(Provider<Credentials> credentialsProvider, String domainName,
+            VrsnMdns api) {
+        this.domainName = domainName;
+        this.api = api;
+        this.credentialsProvider = credentialsProvider;
+    }
+
+    /**
+     * Returns Sorted Set of Resource Record Set to to keep behavior similar to
+     * MDNS web UI.
+     */
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterator() {
+        List<Record> recordList = api.getResourceRecordsList(domainName);
+        return VrsnContentConversionHelper.getSortedSetForDenominator(recordList).iterator();
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterateByName(String name) {
+        checkNotNull(name, "name was null");
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+    }
+
+    protected void put(Filter<ResourceRecordSet<?>> valid, ResourceRecordSet<?> rrset) {
```

Comment:
this indirection isn't helpful, especially when this isn't written yet. also, protected is meaningless as none of these classes should be extended. please remove this.